rp2040-hal/Cargo.toml

```diff
@@ -44,4 +44,4 @@ pio-proc = { git = "https://github.com/rp-rs/pio-rs.git", branch = "main" }
 [features]
 rt = ["rp2040-pac/rt"]
 embassy-traits = ["embassy_traits", "futures"]
-
+alloc = []
```

Comment:
I wonder if the `alloc` feature should be left out, for now:
- It's not clear how it can be used.
  - Enabling the feature causes an error: `error: no global memory allocator found but one is required`
  - Trying to add an allocator leads to a whole rabbit hole of other issues.
- The feature looks perfectly usable without `alloc`. Does the variant of `spawn` with the alloc feature provide some major advantage?

Personally, I'd like the idea of a simple to use `alloc` feature. But it should be a well-documented way of enabling https://doc.rust-lang.org/alloc/ on the RP2040. Allowing for a cleaner implementation of `spawn` is just a little detail, which could easily be added later.

Reply:
I don't think it's really a rabbit hole. I use the alloc spawn and it seems to work fine. it's nice because you can use a closure.

Reply:
Well sure, if you know how alloc works, it may fine:
- enable `alloc` feature
  - `error: no global memory allocator found but one is required`
  - find some allocator crate. I found `static-alloc`
  - add dependency on `static-alloc`
  - add global-allocator declaration to main.rs
    - `error[E0599]: no method named `compare_exchange` found for struct `AtomicUsize` in the current scope`
    - sure, cortex-m0+ doesn't have that
    - Notice that `static-alloc` has a feature `polyfill`
    - get error regarding `#![feature(default_alloc_error_handler)]`
      - add that feature
      - get error regarding nightly
      - switch to nightly compiler
        - Now it compiles, but is probably unsound, as the default critical-section used by atomic-polyfill is not multicore-safe. You also need to add a suitable `custom-impl` for `critical-section`. Probably works fine first, but might have subtle bugs.

Sure, once it works, it's not difficult. But I wonder how long a novice would need to find this out?

Reply:
A lot of those things seem like they're footguns that would happen regardless of whether someone is using multicore or not.
When I was a Rust novice I struggled to even turn features on. I hope it's less of a problem than you think.

Reply:
Probably the main issue is that it's just not documented. But as you already mentioned in other comments, that's true for other parts of the multicore implementation, as well. Your suggestion to defer that to another ticket is ok.

Reply:
Ah, i deleted my reply because it showed up in the review as well, which was confusing. Sorry about that. i'm pasting it here for posterity (should be between jannic's comments.)

> A lot of those things seem like they're footguns that would happen regardless of whether someone is using multicore or not.
> When I was a Rust novice I struggled to even turn features on. I hope it's less of a problem than you think.